facade/src/main/java/org/jboss/pnc/facade/impl/OperationsManagerImpl.java

```diff
@@ -120,9 +120,11 @@ public Tuple _setResult(Base32LongID id, OperationResult result) {
     public DeliverableAnalyzerOperation newDeliverableAnalyzerOperation(
             String milestoneId,
             Map<String, String> inputParams) {
-        ProductMilestone milestone = productMilestoneRepository
-                .queryById(ProductMilestoneMapper.idMapper.toEntity(milestoneId));
-        if (milestone == null) {
+        ProductMilestone milestone = null;
+        if (milestoneId != null) {
+            milestone = productMilestoneRepository.queryById(ProductMilestoneMapper.idMapper.toEntity(milestoneId));
+        }
+        if (milestone == null && milestoneId != null) {
             throw new EmptyEntityException("Milestone with id " + milestoneId + " doesn't exist");
         }
```

Comment:
I think this way it's clearer that the exception is tied to the content of the previous if
```suggestion
        ProductMilestone milestone = null;
        if (milestoneId != null) {
            milestone = productMilestoneRepository.queryById(ProductMilestoneMapper.idMapper.toEntity(milestoneId));
            if (milestone == null) {
                throw new EmptyEntityException("Milestone with id " + milestoneId + " doesn't exist");
            }
        }
```

Reply:
Yes, it's more meaningful your way: refactored, see [7815b4c](https://github.com/project-ncl/pnc/pull/3825/commits/7815b4cfc91b534b4f9dae971bf8ec998179c6b4)